syft/source/directory_resolver.go

```diff
@@ -212,7 +226,9 @@ func (r directoryResolver) requestPath(userPath string) (string, error) {
 func (r directoryResolver) responsePath(path string) string {
 	// always return references relative to the request path (not absolute path)
 	if filepath.IsAbs(path) {
-		return strings.TrimPrefix(path, r.cwd+string(filepath.Separator))
+		// we need to account for the cwd relative to the running process and the given root for the directory resolver
+		prefix := filepath.Clean(filepath.Join(r.cwd, r.cwdRelativeToRoot))
```

Comment:
If someone runs against a linux box on windows I think we run into the issue where this will not do the paths correctly right?

Reply:
I don't think I understand your specific point (yet). The directory resolver is only used against the host's filesystem and not container images. This means that we shouldn't need to worry about cross-platform concerns such as using posix paths for a linux container image being scanned on a windows host. In all cases the scan will be against (and thus relative to) the hosts filesystem paradigm.

Reply:
That's my fault - I thought that this could double as doing a directory resolution within a container as well.